include/lsst/afw/table/io/FitsReader.h

```diff
@@ -15,135 +17,85 @@ namespace lsst { namespace afw { namespace table { namespace io {
-     *  This means Factory and its subclasses should only be constructed as namespace-scope
-     *  objects (so they never go out of scope, and automatically get registered).
-     *
-     *  Subclasses should use this via its derived template class FactoryT.
-     */
-    class Factory {
-    public:
-
-        /// Create a new FITS reader from a cfitsio pointer holder and (optional) input archive and flags
-        virtual PTR(FitsReader) operator()(Fits * fits, PTR(InputArchive) archive, int flags) const = 0;
-
-        virtual ~Factory() {}
-
-        /// Create a factory that will be used when the AFW_TYPE fits key matches the given name.
-        explicit Factory(std::string const & name);
-
-    };
-
-    /**
-     *  @brief Subclass for Factory that constructs a FitsReader.
-     *
-     *  Subclasses should use this by providing a the appropriate constructor and then declaring
-     *  a static data member or namespace-scope FactoryT instance templated over the subclass type.
-     *  This will register the subclass so it can be used with FitsReader::make.
-     */
-    template <typename ReaderT>
-    class FactoryT : public Factory {
-    public:
-
-        /// Create a new FITS reader from a cfitsio pointer holder and (optional) input archive and flags
-        virtual PTR(FitsReader) operator()(Fits * fits, PTR(InputArchive) archive, int flags) const {
-            return boost::make_shared<ReaderT>(fits, archive, flags);
-        }
-
-        /// Create a factory that will be used when the AFW_TYPE fits key matches the given name.
-        explicit FactoryT(std::string const & name) : Factory(name) {}
-
-    };
-
-    /**
-     *  @brief Look for the header key (AFW_TYPE) that tells us the type of the FitsReader to use,
-     *         then make it using the registered factory.
-     */
-    static PTR(FitsReader) make(Fits * fits, PTR(io::InputArchive) archive, int flags);
+    explicit FitsReader(std::string const & persistedClassName);
 
-    /**
-     *  @brief Entry point for reading FITS files into arbitrary containers.
-     *
-     *  This does the work of opening the file, calling FitsReader::make, and then calling
-     *  Reader::read.
-     */
-    template <typename ContainerT, typename SourceT>
-    static ContainerT apply(SourceT & source, int hdu, int flags) {
-        Fits fits(source, "r", Fits::AUTO_CLOSE | Fits::AUTO_CHECK);
-        fits.setHdu(hdu);
-        return apply<ContainerT>(fits, flags);
-    }
-
-    /// @brief Low-level entry point for reading FITS files into arbitrary containers.
     template <typename ContainerT>
-    static ContainerT apply(Fits & fits, PTR(io::InputArchive) archive=PTR(io::InputArchive)(), int flags=0) {
-        PTR(FitsReader) reader = make(&fits, archive, flags);
-        return reader->template read<ContainerT>();
+    static ContainerT apply(afw::fits::Fits & fits, int ioFlags, PTR(InputArchive) archive=nullptr) {
```

Comment:
I think the documentation for the ``apply()`` functions was useful.

In fact, I think it would be useful it could be expanded upon, rather than removed. Coming at this code cold, I'm not sure what ``InputArchive`` is for (and glancing at its documentation doesn't help much). I see that it's used when reading ``Exposure`` and ``Source``, so it'll probably become clear later, but a bit more up-front explanation of what's happening would help.